src/nanotron/config/config.py

```diff
@@ -401,7 +406,7 @@ def get_config_from_dict(
                 SamplerType: lambda x: SamplerType[x.upper()],
             },
             # strict_unions_match=True,
-            strict=True,
+            strict=not igonore_all_unused_keys,
```

Comment:
why do we need this? we'd like it to stay strict if possible?

Reply:
So by default it's true unless we want it to not be strict. Because, sometimes I've to either delete part of the config files by hand or change this argument.

Reply:
Because in my case, I have a model trained with an older dataset config, I cannot load it with the new data_stages config and dataset config. However for evaluation, I don't care about it, I only want to load the model. This is just to add some backward compatibility, and the default value of strict is set to True so it will not trigger any bugs for anyone who is unconscious about this new key.  Btw, we already have similar arguments that can skip unused keys. https://github.com/huggingface/nanotron/blob/4799d247026a8530face56b9daec4b210732fd3b/src/nanotron/config/config.py#L385-L386

Reply:
An example would be: 

The older version of OptimizerArgs: 
```
class OptimizerArgs:
    ...
   adam_beta2: float
```
The latest version of OptimizerArgs:
```
class OptimizerArgs:
    optimizer_factory: Union[SGDOptimizerArgs, AdamWOptimizerArgs]
class AdamWOptimizerArgs:
    ...
    adam_beta2: float
```
For evaluation, I don't think we care about whether we load the optimizer state correctly. You are right, we'd like it to stay strict for training, and this argument should only be used for evaluation.